src/main/java/com/spotify/docker/client/DockerConfigReader.java

```diff
@@ -0,0 +1,117 @@
+/*-
+ * -\-\-
+ * docker-client
+ * --
+ * Copyright (C) 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.docker.client;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.docker.client.messages.RegistryAuth;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+import org.glassfish.jersey.internal.util.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DockerConfigReader {
+  private static final Logger LOG = LoggerFactory.getLogger(DockerConfigReader.class);
+
+  private static final ObjectMapper MAPPER = ObjectMapperProvider.objectMapper();
+
+
+  public RegistryAuth fromComfig(Path configPath, String serverAddress) throws IOException {
+    return parseDockerConfig(configPath, serverAddress).build();
+  }
+
+  public RegistryAuth.Builder parseDockerConfig(final Path configPath, String serverAddress)
+      throws IOException {
+    checkNotNull(configPath);
+    final RegistryAuth.Builder authBuilder = RegistryAuth.builder();
+    final JsonNode authJson = this.extractAuthJson(configPath);
+
+    if (isNullOrEmpty(serverAddress)) {
+      final Iterator<String> servers = authJson.fieldNames();
+      if (servers.hasNext()) {
+        serverAddress = servers.next();
```

Comment:
This will return the first credential in the file. We need the one, if any, that equals `serverAddress`.